### ISSUE TITLE
S3 Adapter - make public ACL configurable

### ISSUE THREAD

**Issue by FireFart** [kjvarga/sitemap_generator#365]
Currently the public ACL on S3 storage is hardcoded:
https://github.com/kjvarga/sitemap_generator/blob/28dd22c48597f8b2a44cbc7e20685e84975fe75b/lib/sitemap_generator/adapters/s3_adapter.rb#L50

When a bucket is configured as private with no public access, setting a public ACL on an object will fail. In our case we fetch the sitemap via the s3 protocol and send it via nginx back to the requestor. In this case there is no need for this to be public.

Any chance to make this configurable?

**Comment by FireFart** [comment]
Same on the AWSS3Adapter:  https://github.com/kjvarga/sitemap_generator/blob/28dd22c48597f8b2a44cbc7e20685e84975fe75b/lib/sitemap_generator/adapters/aws_sdk_adapter.rb#L35

This needs to be set to `private` in this case

**Comment by nplusp** [comment]
I would like to +1 for the Google Cloud Storage adapter as well. Having a uniform bucket-level enabled, the GCS gem does not accept the ACL and raises an error:
```
Google::Cloud::InvalidArgumentError: invalid: Cannot insert legacy ACL for an object when uniform bucket-level access is enabled. Read more at https://cloud.google.com/storage/docs/uniform-bucket-level-access
/usr/local/bundle/gems/google-cloud-storage-1.21.1/lib/google/cloud/storage/service.rb:636:in `rescue in execute'
/usr/local/bundle/gems/google-cloud-storage-1.21.1/lib/google/cloud/storage/service.rb:633:in `execute'
/usr/local/bundle/gems/google-cloud-storage-1.21.1/lib/google/cloud/storage/service.rb:311:in `insert_file'
/usr/local/bundle/gems/google-cloud-storage-1.21.1/lib/google/cloud/storage/bucket.rb:1269:in `create_file'
/usr/local/bundle/gems/sitemap_generator-6.1.2/lib/sitemap_generator/adapters/google_storage_adapter.rb:34:in `write'
/usr/local/bundle/gems/sitemap_generator-6.1.2/lib/sitemap_generator/sitemap_location.rb:166:in `write'
/usr/local/bundle/gems/sitemap_generator-6.1.2/lib/sitemap_generator/builder/sitemap_file.rb:139:in `write'
/usr/local/bundle/gems/sitemap_generator-6.1.2/lib/sitemap_generator/builder/sitemap_index_file.rb:139:in `write_first_sitemap'
/usr/local/bundle/gems/sitemap_generator-6.1.2/lib/sitemap_generator/builder/sitemap_index_file.rb:105:in `finalize!'
/usr/local/bundle/gems/sitemap_generator-6.1.2/lib/sitemap_generator/link_set.rb:474:in `finalize_sitemap_index!'
/usr/local/bundle/gems/sitemap_generator-6.1.2/lib/sitemap_generator/link_set.rb:345:in `finalize!'
/usr/local/bundle/gems/sitemap_generator-6.1.2/lib/sitemap_generator/link_set.rb:41:in `create'
/usr/local/bundle/gems/sitemap_generator-6.1.2/lib/sitemap_generator.rb:42:in `method_missing'
/usr/src/app/config/sitemap.rb:15:in `run'
/usr/local/bundle/gems/sitemap_generator-6.1.2/lib/sitemap_generator/interpreter.rb:78:in `instance_eval'
/usr/local/bundle/gems/sitemap_generator-6.1.2/lib/sitemap_generator/interpreter.rb:78:in `run'
/usr/local/bundle/gems/sitemap_generator-6.1.2/lib/sitemap_generator/tasks.rb:51:in `block (2 levels) in <main>'
/usr/local/bundle/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'
```

**Comment by kjvarga** [comment]
This has been addressed in v6.2.0 for the `S3Adapter`.  If you'd like to open PRs for the other adapter(s) that would be great.

**Comment by codyrobbins** [comment]
For anyone that needs it until it’s merged, I've created a pull request for a similar modification for `AwsSdkAdapter` at https://github.com/kjvarga/sitemap_generator/pull/394.
